docs/design/windows-images.md

```diff
@@ -0,0 +1,164 @@
+`DefaultMobyVersion` | Version of Moby to install if applicable
+`WindowsPatchList` | List of public patches to install using Windows Update, by [KB number](https://support.microsoft.com/en-us/help/4464619)
+`WindowsPrivatePatchList` | After public patches are applied, this list of private patches will be downloaded and applied. This will turn test signing on.
+
+
+#### Parameters for shipped OS versions
+
+These parameters can only be used when starting from an existing Windows OS image available from the Azure marketplace. It should not have any container runtime installed or it may cause conflicts.
+
+Parameter  | Description
+-----------|-------------
+`WindowsBaseImage` | Existing Azure image to boot up and customize. Mutually exclusive with `WindowsIsoUrl`. Example: `MicrosoftWindowsServer:WindowsServerSemiAnnual:Datacenter-Core-1809-smalldisk:1809.0.20181107`
+
+#### Parameters for prerelease OS versions
+
+These parameters cannot be used when `WindowsBaseImage` is set. This requires running Packer to boot a new VM, and run an unattended Windows setup. Once the unattended setup is complete, the remaining steps will be the same.
+
+Parameter  | Description
+-----------|-------------
+`WindowsIsoUrl` | ISO image to build from. Can be an Azure storage blob requiring login. Mutually exclusive with `WindowsIsoUrl`
+`NanoServerName` | Full name `mcr.microsoft.com/.../nanoserver-insider`
+
+ISO builds will need a few more files in this repo for Windows VHD creation:
+
+- `unattend.xml` - needed to configure default user, drive layout, and trial license keys
+- `windowsserverinsider.json` - Packer configuration for the VM build.
+- Some additional scripts may be called from the packer JSON if needed to install the Azure agent and other necessary steps.
+
+### Deployment Phase
+
+Once a VM is deployed, the right configuration should be applied. This is handled by a custom script included in the ARM template for that VM.
+
+It will:
+
+1. Finish installing the desired Kubernetes, Moby/Docker/ContainerD, and network plugin versions. The user will provide these through the AKS-Engine [Cluster Definition](docs\topics\clusterdefinitions.md)
+2. Create or update configuration files for the above. Many of these have interdependencies based on file paths, binary names, or other configurations such as kubelet parameters that aren't known until deployment time.
+3. Start the necessary services
+4. Join the Kubernetes cluster
+
+## AKS-Engine Apimodel Changes
+
+These parameters are defined in [Cluster Definition](docs\topics\clusterdefinitions.md), and will have impact on what needs to be put in the Windows images and the configuration steps needed as they're deployed. Some of these are not used as-is, but instead are used to compute other values such as a path to a tarball on the acsmirror CDN. These ones should be checked for cached downloads before trying the acsmirror CDN since they may be in the VHD already.

```

Comment:
checking for the cached downloads before trying the mirror sounds like it could be complicated to debug if something when wrong.  Is it possible to say make parameters override (if provided)?